docs/developer-docs/security/rust-canister-development-security-best-practices.md

```diff
@@ -10,20 +10,32 @@ This document contains canister development best practices for both Motoko and R
 
 ## Smart contracts canister control
 
-### Use a decentralized governance system like SNS to make a canister have a decentralized controller
+### Use a decentralized governance system like SNS to put dapps under decentralized control
 
 #### Security concern
 
-The controller of a canister can change or update the canister whenever they like. If a canister e.g. stores assets such as ICP, this effectively means that the controller can steal these by updating the canister and transfer the cycles to their account.
+The controller of centralized components of a dapp (such as centrally controlled canisters or cloud services) can change or update these components whenever they like. If a canister or off-chain component e.g. holds assets such as ICP, ckBTC or ckETH on the user's behalf, this effectively means that the controller could decide at any time to steal these funds by e.g. updating the canister and transferring the assets to their account.
 
-#### Recommendation
+Furthermore, the controller of canisters serving web content (such as e.g. the [asset canister](../../references/asset-canister)) could maliciously modify the web application to e.g. steal user funds or perform security sensitive actions on the user's behalf. For example, if [Internet Identity](../../developer-docs/integrations/internet-identity/overview) is used, the user principal's private key for the given origin is stored in the browser storage, and a malicious app can therefore fully control the private key and the user's session and any assets controlled by that key.  
+
+#### Recommendations
+
+The following list starts with recommendations for dapps under centralized control, and then provides increasingly stronger recommendations to strengthen the dapp's security in terms of decentralization. 
```

Comment:
I stumbled over this a bit.. I guess we want to say 1) we start with a centralized setting and go to increasinly decetnralized ones and 2) the decentralized is more favorable from a security pov?